app/controllers/hyrax/batches_controller.rb

```diff
@@ -3,7 +3,8 @@ class BatchesController < ApplicationController
     before_action :check_permissions
 
     def index
-      @batches = Batch.all.map { |batch| BatchPresenter.for(object: batch) }
+      @batches = batch_query(start, stop)
+      @batches_length = Batch.all.length
```

Comment:
Please use `Batch.count` instead of `Batch.all.length` to avoid loading all batches.